diff -ruN ./include/caffe/util/io.hpp ../caffe_win_10052015/include/caffe/util/io.hpp
--- ./include/caffe/util/io.hpp	Sun May 10 17:14:55 2015
+++ ../caffe_win_10052015/include/caffe/util/io.hpp	Sun May 10 18:07:12 2015
@@ -1,7 +1,11 @@
 #ifndef CAFFE_UTIL_IO_H_
 #define CAFFE_UTIL_IO_H_
 
+#ifdef _MSC_VER
+#include "wincaffe_misc.h"
+#else
 #include <unistd.h>
+#endif
 #include <string>
 
 #include "google/protobuf/message.h"
diff -ruN ./src/caffe/common.cpp ../caffe_win_10052015/src/caffe/common.cpp
--- ./src/caffe/common.cpp	Sun May 10 17:14:55 2015
+++ ../caffe_win_10052015/src/caffe/common.cpp	Sun May 10 18:18:55 2015
@@ -5,6 +5,10 @@
 #include "caffe/common.hpp"
 #include "caffe/util/rng.hpp"
 
+#ifdef _MSC_VER
+#include <process.h> 
+#endif
+
 namespace caffe {
 
 shared_ptr<Caffe> Caffe::singleton_;
@@ -36,7 +40,9 @@
   // Google logging.
   ::google::InitGoogleLogging(*(pargv)[0]);
   // Provide a backtrace on segfault.
+#ifndef _MSC_VER
   ::google::InstallFailureSignalHandler();
+#endif
 }
 
 #ifdef CPU_ONLY  // CPU-only Caffe.
diff -ruN ./src/caffe/layers/bnll_layer.cu ../caffe_win_10052015/src/caffe/layers/bnll_layer.cu
--- ./src/caffe/layers/bnll_layer.cu	Sun May 10 17:14:55 2015
+++ ../caffe_win_10052015/src/caffe/layers/bnll_layer.cu	Sun May 10 17:53:36 2015
@@ -6,7 +6,11 @@
 
 namespace caffe {
 
+#ifdef _MSC_VER
+#define kBNLL_THRESHOLD 50.
+#else
 const float kBNLL_THRESHOLD = 50.;
+#endif
 
 template <typename Dtype>
 __global__ void BNLLForward(const int n, const Dtype* in, Dtype* out) {
diff -ruN ./src/caffe/solver.cpp ../caffe_win_10052015/src/caffe/solver.cpp
--- ./src/caffe/solver.cpp	Sun May 10 17:15:05 2015
+++ ../caffe_win_10052015/src/caffe/solver.cpp	Sun May 10 18:12:39 2015
@@ -11,6 +11,11 @@
 #include "caffe/util/math_functions.hpp"
 #include "caffe/util/upgrade_proto.hpp"
 
+#ifdef _MSC_VER
+#include <stdio.h>
+#define snprintf _snprintf
+#endif
+
 namespace caffe {
 
 template <typename Dtype>
diff -ruN ./src/caffe/util/db.cpp ../caffe_win_10052015/src/caffe/util/db.cpp
--- ./src/caffe/util/db.cpp	Sun May 10 17:14:55 2015
+++ ../caffe_win_10052015/src/caffe/util/db.cpp	Sun May 10 18:16:09 2015
@@ -3,6 +3,11 @@
 #include <sys/stat.h>
 #include <string>
 
+#ifdef _MSC_VER
+#include <direct.h>
+#define mkdir(name,mode) _mkdir(name)
+#endif
+
 namespace caffe { namespace db {
 
 const size_t LMDB_MAP_SIZE = 1099511627776;  // 1 TB
diff -ruN ./src/caffe/util/io.cpp ../caffe_win_10052015/src/caffe/util/io.cpp
--- ./src/caffe/util/io.cpp	Sun May 10 17:15:05 2015
+++ ../caffe_win_10052015/src/caffe/util/io.cpp	Sun May 10 17:37:05 2015
@@ -48,7 +48,11 @@
 }
 
 bool ReadProtoFromBinaryFile(const char* filename, Message* proto) {
+#ifdef _MSC_VER
+  int fd = open(filename, O_RDONLY | O_BINARY);
+#else
   int fd = open(filename, O_RDONLY);
+#endif
   CHECK_NE(fd, -1) << "File not found: " << filename;
   ZeroCopyInputStream* raw_input = new FileInputStream(fd);
   CodedInputStream* coded_input = new CodedInputStream(raw_input);
diff -ruN ./src/caffe/util/math_functions.cpp ../caffe_win_10052015/src/caffe/util/math_functions.cpp
--- ./src/caffe/util/math_functions.cpp	Sun May 10 17:14:55 2015
+++ ../caffe_win_10052015/src/caffe/util/math_functions.cpp	Sun May 10 18:08:17 2015
@@ -7,6 +7,10 @@
 #include "caffe/util/math_functions.hpp"
 #include "caffe/util/rng.hpp"
 
+#ifdef _MSC_VER
+#include "wincaffe_misc.h"
+#endif
+
 namespace caffe {
 
 template<>
